src/components/content-loading-segment.tsx

```diff
@@ -0,0 +1,319 @@
+import React from 'react'
+import {colors} from '../styles'
+import {View, StyleSheet, ViewProps} from 'react-native'
+import {
+  Placeholder,
+  PlaceholderMedia,
+  PlaceholderLine,
+  Fade,
+} from 'rn-placeholder'
+
+export enum ContentLoadingSegmentSize {
+  Small,
+  Large,
+}
+
+interface Props extends ViewProps {
+  size: ContentLoadingSegmentSize
+}
+
+export const ContentLoadingSegment = ({size}: Props) => {
+  const lineHeightThick = 16
+  const lineHeightThin = 12
+  const circleDiameter = 36
+
+  if (size === ContentLoadingSegmentSize.Small) {
+    return (
```

Comment:
I think to make this code nicer, I would suggest breaking it into smaller components.

```
if (size === ContentLoadingSegmentSize.Small) {
  return <SmallLoadingSegment />
} else if ... {
  return <LargeLoadingSegment />
}
```